### PR TITLE
Run simplenote

### PR DESCRIPTION
**Pull request by beaucollins**
At this point `node-simperium`s only reason for life is `simplenote-electron`.

To help remove testing friction this adds a bin script that:

1. Checks out Automattic/simplenote-electron to `tmp/simplenote-electron`
2. `npm install` and `npm link ../../` so that the current `node-simperium` source you are working on will be used by `simplenote-electron`
3. Boots the `simplenote-electron` web server and opens your browser.

Now when asking for a review you should hopefully be able to start with:

> Run `./bin/simplenote`

### How to test

- Run `./bin/simplenote`
- Wait a bit while
  - git clones the repo
  - npm installs the deps
  - weback-dev-server transpiles the JS
- When it's down, your browser should be opened (on macOS)

If on linux or Windows you will have to open http://localhost:4000 yourself.

### Future Improvements

- If `node-simperium` is to live on in its own repository Puppeteer could be leveraged to fully automate the browser testing.
- The source in `simplenote-electron/lib/simperium` could make its way into this library so the interface between the browser and `node-simperium` could be more easily tested using a JSDom style library.

### CODE REVIEW

**Comment by dmsnell** [comment]
This seems both exciting and it also makes me nervous. Would we be able to accomplish something almost as helpful if we made it clearer in the docs how to use `npm link` and run Simplenote (or any other app) with this?

That way we wouldn't intrinsically tie the two together or duplicate the run script here that exists there…

**Comment by beaucollins** [comment]
> Would we be able to accomplish something almost as helpful if we made it clearer in the docs how to use npm link and run Simplenote (or any other app) with this?

Probably? The goal for me was to get this to a stage where it would be driven 100% by Puppeteer and then could live on the CI.

But I'm also aware the library could easily be moved _into_ the `simplenote-electron` repository to make all of this easier.

I guess the main goal was to make it _really easy_ for someone to test this without ever having seen or touched `simplenote-electron`. If one already has a dev setup for `simplenote-electron` then ideally they would understand how to use `npm link` and they can ignore usage of this script.

TL;DR this probably deserves a README about why you would want to use it.

**Comment by dmsnell** [comment]
I'm thinking that for now it may be best to leave this be. I'm open to the idea of merging this code into `simplenote-electron`, but as you raised in a private discussion, there is also value in keeping them separate.

Pupeteer or automated browser testing seems more appropriate for a project like `simplenote-electron` and I'd prefer we leave it where the browser actually comes into play, meaning not here.

I'm nervous about adding the complexity to this project especially with how we're starting a new period of churn which might drastically change things. Maybe that means testing is more important, but also makes it harder to change. I'm leaning towards "we're figuring out what needs to happen before we enforce behaviors"
